src/sentry/static/sentry/app/components/events/eventEntry.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+
+import Breadcrumbs from 'app/components/events/interfaces/breadcrumbs';
+import Csp from 'app/components/events/interfaces/csp';
+import DebugMeta from 'app/components/events/interfaces/debugMeta';
+import Exception from 'app/components/events/interfaces/exception';
+import Generic from 'app/components/events/interfaces/generic';
+import Message from 'app/components/events/interfaces/message';
+import Request from 'app/components/events/interfaces/request';
+import Spans from 'app/components/events/interfaces/spans';
+import Stacktrace from 'app/components/events/interfaces/stacktrace';
+import Template from 'app/components/events/interfaces/template';
+import Threads from 'app/components/events/interfaces/threads';
+import {Organization, Project, SharedViewOrganization} from 'app/types';
+import {Entry, EntryType, Event, EventTransaction} from 'app/types/event';
+
+type Props = {
+  entry: Entry;
+  projectSlug: Project['slug'];
+  event: Event;
+  organization: SharedViewOrganization | Organization;
+  isShare: boolean;
+};
+
+function EventEntry({entry, event, projectSlug, organization}: Props) {
+  switch (entry.type) {
+    case EntryType.EXCEPTION: {
+      const {data, type} = entry;
+      return <Exception type={type} event={event} data={data} projectId={projectSlug} />;
```

Comment:
Would it be easier to destructure the entire `entity` above and write these as 

```tsx
case EntryType.EXCEPTION: {
	return <Exception {...{type, event, data, projectId}} />;
}
case EntryType.MESSAGE: {
  	return <Message {...{data}} />;
}
	
case EntryType.REQUEST: {
	  return <Request {...{type, event, data}} />;
}
```

just my thoughts, but I think that is easier to read.

Reply:
I prefer to have it separated 🤷‍♀️ but I don't have strong feelings .. @getsentry/app-frontend please what do you think about it?

Reply:
The props vary so much that I would prefer them separated as well, but I don't have strong feelings either way.

Reply:
Cool, let's leave it 👍